forte/data/ontology/top.py

```diff
@@ -686,5 +688,169 @@ def get(
+    Regular grids with a grid configuration.
+
+    Args:
+        pack: The container that this grids
+            will be added to.
+        height_n_width: A tuple represents the number of grid cell per column and
+            the number of grid cell per row.
+    """
+
+    def __init__(
+        self,
+        pack: PackType,
+        height: int,
+        width: int,
+        image_payload_idx: Optional[int] = None,
+    ):
+
+        if height <= 0 or width <= 0:
+            raise ValueError(
+                f"height({height}) and "
+                f"width({width}) both must be larger than 0"
+            )
+        self.height = height
+        self.width = width
+        self._image_payload_idx = image_payload_idx
+        super().__init__(pack)
+
+    def get_grid_cell(
+        self, h_idx: int, w_idx: int, image_payload_idx: Optional[int] = None
+    ):
+        """
+        Get the array data of a grid cell from image of the image payload index.
+        The array is the same size of the image. The array entries that are not
+        within the grid cell will be zeros, and the array entries that are
+        within the grid cell will be preserved.
+
+
+        Args:
+            h_idx: the index of the grid cell of the first dimension.
+            w_idx: the index of the grid cell of the second dimension.
+            image_payload_idx: An integer that represents the index of
+                the image in the payloads. If it's not None, it will use the
+                image from the corresponding payload. Otherwise,
+                it will use the image
+                from the payload of ``self._image_payload_idx``.
+
+        Raises:
+            ValueError: ``h_idx`` is out of the range specified by ``height_n_width``.
+            ValueError: ``w_idx`` is out of the range specified by ``height_n_width``.
+
+        Returns:
+            numpy array that represents the grid cell.
+        """
+        if not 0 <= h_idx < self.height:
+            raise ValueError(
+                f"input parameter h_idx ({h_idx}) is"
+                "out of scope of h_idx range"
+                f" {(0, self.height)}"
+            )
+        if not 0 <= w_idx < self.width:
+            raise ValueError(
+                f"input parameter w_idx ({w_idx}) is"
+                "out of scope of w_idx range"
+                f" {(0, self.width)}"
+            )
+
+        if image_payload_idx is None:
+            if self._image_payload_idx is None:
+                raise ValueError(
+                    "There is not image_payload_idx associate "
+                    "with the Grids instance neither"
+                    " image_payload_idx in the ``get_grid_cell`` "
+                    "function parameters. Please make sure either "
+                    "one of them is valid."
+                )
+            image_payload_idx = self._image_payload_idx
+
+        img_arr = self.pack.get_image_array(image_payload_idx)
+        c_h, c_w = (
```

Comment:
I feel like some numbers should be pre-computed and stored, like `c_h` and `c_w`.